EIPS/eip-sss_serialization.md

```diff
@@ -0,0 +1,276 @@
+  - the current serialization scheme being used in Eth2.0
+- SSS aka Streamable Simple Serialize (working title):
+  - https://github.com/ethereum/bimini/blob/master/spec.md
+  - an experimental serialization scheme developed specifically for the networking needs of the Ethereum protocol.
+
+
+#### RLP
+
+RLP is well known and widely used.
+
+- **Language Support**
+  - Strong: Established RLP libraries available across most languages
+- **Strongly Typed**
+  - No: however strong typing is added at the implementation level for most RLP libraries.
+- **Compact / Efficient**
+  - Medium:
+    - No native support for fixed length byte strings results in extra bytes for length prefix
+    - No native support for fixed length arrays results in extra bytes for length prefix
+    - No layer-2 support for compact integer serialization
+- **Streamable**
+  - Yes
+
+
+#### SSZ
+
+SSZ was created as part of the Eth2.0 research.  It was designed to be simple
+and low overhead.
+
+- **Language Support**
+  - Low: SSZ implementations are being created as part of the Eth2.0 efforts.  None are well established.
+- **Strongly Typed**
+  - Yes
+- **Compact / Efficient**
+  - Poor:
+    - No support for compact integer serialization
+    - Length prefixes are 32-bit, typically resulting in multiple superflous empty bytes
+    - Containers are size prefixed with 32-bit values, all of which are superflous
+    - No support for compact integer serialization
+- **Streamable**
+  - No:
+    - Size prefixing container types prevents streaming
+
+
+#### SSS
+
+SSS was created to specifically address the Ethereum network needs.
+
+- **Language Support**
+  - Bad: Only one experimental implementation in python https://github.com/ethereum/bimini
+- **Strongly Typed**
+  - Yes
+- **Compact / Efficient**
+  - High
+    - Near zero superfluous bytes for most schemas
+- **Streamable**
+  - Yes
+
+## Side-by-Side Comparison
+
+Empirical tests were done to shows how SSS performs with respect to RLP and SSZ for
+the following data structures with respect to their serialized sizes.
+
+- Blocks
+- Headers
+- Transactions
+- Receipts
+- Logs
+- Accounts
+- State Trie Nodes (of depths 0-9)
+- Discovery Ping
+- Discovery Pong
+- Discovery FindNode
+- Discovery Neighbours
+
+In every case, SSS serialization resulted in a smaller serialized
+representation than it's counterparts in either RLP or SSZ.
+
+The comparison to SSZ is less relevant to the Eth1.0 network, but tests are
+planned against the Eth2.0 data structures.
+
+
+### SSS vs RLP Summary
+
+The comparison to RLP shows that there are multiple places where SSS could
+reduce the amount of bandwidth used by Ethereum clients.
+
+- ~3% reduction in block size
+- ~50% reduction in receipt size
+- ~5% reduction in account size (state sync)
+- ~2% reduction in trie-node size
+
+Specific to the discovery protocol:
+
+- ~8% reduction for ping
+- ~4% reduction for pong
+- ~4% reduction for find-nodes
+- ~4% reduction for neighbors
```

Comment:
So far all of these numbers seem too small to justify "having yet another distinct serialization protocol", with the exception of receipt size. Why are receipts currently too big? Also keep in mind that eth1 receipts != eth2 receipts; eth2 receipts will probably be encoded with some ABI-like encoding rather than RLP.

Reply:
Almost all of the SSS efficiency is from two optimizations

- the `scalar<N>` type from SSS uses `LEB128` 
- using `scalar<N>` for length prefixes since it auto-scales it's size at the cost of one bit per byte.

After writing all of this up, I think we can actually reap these benefits under RLP since RLP doesn't actually define any serialization beyond bytes and lists, and all of the *rich* objects are actually layer-2 on top of RLP.  Something similar is probably doable with SSZ.

Reply:
Quick test shows that the average size savings for receipts drops to `~8%` with a naive introduction of a `scalar<N>` type in the `pyrlp` library.